### PR TITLE
Implement junit testing

### PR DESCRIPTION
**Pull request by IsaccBarker**
Now we don't have to wait until we get an actual robot to test, and debug on. We can run tests locally, on a machine that isn't running Android.

Ideally, everyone should write tests for things that are applicable.

Testing is implemented using JUnit 5, and you can see some examples on their documentation, or the tests in the testing source tree.

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/environment/EnvironmentMap.java [hunk 1]
```diff
@@ -81,5 +75,25 @@ public Entity get_active_object() throws NoActiveEntityException {
...
+        }
+    }
+
+    /** Sets all the enties; doesn't append. */
+    public void setAllEntities(ArrayList<Entity> new_entities) {
+        entities = new_entities;
+    }
+
+    public void foo() {}
 }
```
Comment (DragonDev07) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/environment/EnvironmentMap.java:98]: Why emoty function?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/environment/EnvironmentMapError.java [hunk 2]
```diff
@@ -1,3 +1,5 @@
+package org.firstinspires.ftc.teamcode.environment;
+
 class NoActiveEntityException extends Exception {
     public NoActiveEntityException() { }
```
Comment (DragonDev07) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/environment/EnvironmentMapError.java:4]: Empty Function?

File: TeamCode/src/test/java/org/firstinspires/ftc/teamcode/environment/EntityTest.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package org.firstinspires.ftc.teamcode.environment.test;
+
+class EntityTest { }
```
Comment (DragonDev07) [inline TeamCode/src/test/java/org/firstinspires/ftc/teamcode/environment/EntityTest.java:3]: EMPTY! :D

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/environment/EnvironmentMap.java (hunk 1 above)
Comment (IsaccBarker) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/environment/EnvironmentMap.java:98]: sorry, that was a weird testing function i made. removing.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/environment/EnvironmentMapError.java (hunk 2 above)
Comment (IsaccBarker) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/environment/EnvironmentMapError.java:4]: That's actually just a constructor, that constructs nothing. Think of it like a subtype, and thats just a declaration.

File: TeamCode/src/test/java/org/firstinspires/ftc/teamcode/environment/EntityTest.java (hunk 3 above)
Comment (IsaccBarker) [reply TeamCode/src/test/java/org/firstinspires/ftc/teamcode/environment/EntityTest.java:3]: lol, that's just for the future. now that I think about it, I should probably remove it. Thanks :D

**Review by DragonDev07 — APPROVED** [review]
Sweet, Maybe fix that empty file?

**Comment by IsaccBarker** [comment]
Yep! About to commit that....

**Pull request merged**